superset-frontend/src/components/DatabaseSelector/changeDataBase/useChangeDataBase.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect, useRef } from 'react';
+import { FetchSchemas } from '../fetchSchemas';
+import { OnSelectChange } from '../onSelectChange';
+import { factoryChangeDataBase } from './factoryChangeDataBase';
+
+export const useChangeDataBase = ({
+  setSchemaOptions,
+  fetchSchemas,
+  onSchemaChange,
+  onDbChange,
+  onSelectChange,
+}: {
+  setSchemaOptions: (options: any[]) => void;
+  fetchSchemas: React.MutableRefObject<FetchSchemas>;
+  onSchemaChange?: (arg0?: any) => {};
+  onDbChange?: (db: any) => void;
+  onSelectChange: React.MutableRefObject<OnSelectChange>;
+}) => {
+  const changeDataBase = useRef(
+    factoryChangeDataBase({
+      setSchemaOptions,
+      fetchSchemas,
+      onSchemaChange,
+      onDbChange,
+      onSelectChange,
+    }),
+  );
+  useEffect(() => {
+    changeDataBase.current = factoryChangeDataBase({
+      setSchemaOptions,
+      fetchSchemas,
+      onSchemaChange,
+      onDbChange,
+      onSelectChange,
+    });
+  }, [
+    fetchSchemas,
+    onDbChange,
+    onSchemaChange,
+    onSelectChange,
+    setSchemaOptions,
+  ]);
+
+  return changeDataBase;
```

Comment:
Also, the caller of this hook has to do `changeDataBase.current()`. If we continue using a ref for this, it should return `changeDataBase.current` instead.